File: 노영주/32주차/요격시스템.js
function solution(targets) {
  targets.sort((a, b) => a[0] - b[0] || a[1] - b[1]);

  let answer = 1;
  let currE = targets[0][1];

  for (let i = 1; i < targets.length; i++) {
    const [s, e] = targets[i];

    if (s >= currE) {
      answer++;
      currE = e;
    } else {
      currE = Math.min(currE, e);
    }
  }

  return answer;
}

File: 윤호선/32주차/요격시스템.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool comp(const vector<int>& a, const vector<int>& b){
    if(a[0]==b[0])return a[1]<b[1];
    return a[0]<b[0];
}
int solution(vector<vector<int>> targets) {
    int answer = 1;
    
    sort(targets.begin(),targets.end(),comp);
    
    int y=targets[0][1];
    for(int i=1;i<targets.size();i++){
        if(targets[i][0]<y){
            if(targets[i][1]<y)y=targets[i][1];
        }
        else{
            answer++;
            y=targets[i][1];
        }
    }
    return answer;
}
